packages/block-library/src/embed/index.js

```diff
@@ -128,7 +132,22 @@ export function getEmbedEdit( title, icon ) {
 				if ( includes( html, 'class="wp-embedded-content" data-secret' ) ) {
 					// If this is not the WordPress embed block, transform it into one.
 					if ( this.props.name !== 'core-embed/wordpress' ) {
-						this.props.onReplace( createBlock( 'core-embed/wordpress', { url } ) );
+						this.props.onReplace(
+							createBlock(
```

Comment:
Not really a thing to fix in this PR, but I thought i'd mention that it'd be tidier if `maybeSwitchBlock` returned the value of createBlock so that it becomes purer and somewhat side-effect free (depending on what createBlock does).

Reply:
Yes, too many side effects caused this problem. Separating the preview->attribute generation from the code that set the attributes let us fix this, same should be done here. Once this is in I'll do it as a `chore` PR.